tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,455 @@
+
+    @pytest.mark.skipif(
+        not is_nonempty_dir(REAL_PROFET_DATA_DIR), reason="Real profet data is required.",
+    )
+    @pytest.mark.timeout(15)
+    def test_mock_load_data_fixture_when_real_data_available(
+        self, tmp_path_factory, mock_load_data
+    ):
+        """Test that the mock_load_data fixture returns real data when the input dir is non-empty,
+        and that the means, dtypes, shapes, etc match between the real and fake data.
+        """
+        # NOTE: Need to re-import, because the `mock_load_data` fixture modifies that function in the
+        # module. If we imported the function at the top of this test module, we would get the original
+        # version.
+
+        model_config = self.Task.ModelConfig()
+
+        real_input_dir: Path = REAL_PROFET_DATA_DIR
+        real_x, real_y, real_c = model_config.load_data(real_input_dir)
+
+        fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+        fake_x, fake_y, fake_c = model_config.load_data(fake_input_dir)
+        assert real_x.shape == fake_x.shape
+        assert real_y.shape == fake_y.shape
+        assert real_c.shape == fake_c.shape
+
+        assert real_x.dtype == fake_x.dtype
+        assert real_y.dtype == fake_y.dtype
+        assert real_c.dtype == fake_c.dtype
+
+        assert 0 <= real_x.min() and 0 <= fake_x.min()
+        assert real_x.max() <= 1 and fake_x.max() <= 1
+
+        min_y, max_y = y_min[model_config.benchmark], y_max[model_config.benchmark]
+        assert min_y <= real_y.min() and min_y <= fake_y.min()
+        assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+        min_c, max_c = c_min[model_config.benchmark], c_max[model_config.benchmark]
+        assert min_c <= real_c.min() and min_c <= fake_c.min()
+        assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+    @pytest.mark.timeout(30)
+    @pytest.mark.parametrize("device_str", _devices)
+    def test_attributes(
+        self,
+        profet_train_config: MetaModelConfig,
+        profet_input_dir: Path,
+        device_str: str,
+        tmp_path_factory,
+    ):
+        """Simple test: Check that a task can be created and that its attributes are used and set
+        correctly.
+        """
+        max_trials = 123
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
+        device = torch.device(device_str)
+        task = self.Task(
+            max_trials=max_trials,
+            model_config=profet_train_config,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+            device=device,
+        )
+        assert task.max_trials == max_trials
+        assert task.input_dir == profet_input_dir
+        assert task.checkpoint_dir == checkpoint_dir
+        assert isinstance(task.device, torch.device)
+        assert task.device == device
+        assert task.device.type == device_str
+        # NOTE: The class attributes might differ per task, but that's ok.
```

Comment:
If we set the task config inside the subclass for tests then we can test these attributes as well. Like we do for algorithm generic tests.

Reply:
Hm I'm not sure I understand what you mean here, would you mind explaining it again?
You'd like tests about the class attributes of `MetaModelConfig`?

Reply:
Yeah, and if it changes from one task to another, that can be handled by setting the expected modelconfig as an attribute of the test subclass of the specific task.

Reply:
There's probably something I'm missing here, or some confusion. Are you suggesting to tests the (constant) class attributes?
If so, then I disagree: wouldn't that amount to doing `task.ModelConfig.<some_constant> == <value_copy_pasted_from_task.ModelConfig>` ?

Reply:
Oh, no, values that can be configured by user. Are they all constant?

Reply:
They are all constant.

Reply:
Wait no, that's not 100% correct: Most of the attributes of the task's `ModelConfig` inner class are constant class attributes, but some can be changed (`num_steps`, etc). However I don't believe it's necessary to add the type of tests you're suggesting here, because the task just receives the object, and sets it as an attribute on `self`. This test here is sufficient for that, since it checks that the attribute on `self` is equal to the value passed to the constructor. This also checks that the attributes of that config are the same as the constructor arguments that were passed.

Reply:
Oh right, makes sense. Thanks!